### PR TITLE
Remove unnecessary null pointer checks

### LINKED ISSUE #595: Remove unnecessary null pointer checks
[An extra null pointer check is not needed](http://pubs.opengroup.org/onlinepubs/9699919799/functions/free.html "Do I need to check for NULL before free(p)?") in functions like the following.
* [uv__getaddrinfo_done](https://github.com/libuv/libuv/blob/764877fd9e4ea67c0cbe27bf81b2b294ed33b0f5/src/win/getaddrinfo.c#L90)
* [uv_dlclose](https://github.com/libuv/libuv/blob/764877fd9e4ea67c0cbe27bf81b2b294ed33b0f5/src/unix/dl.c#L41)
* [uv_set_process_title](https://github.com/libuv/libuv/blob/764877fd9e4ea67c0cbe27bf81b2b294ed33b0f5/src/unix/openbsd.c#L163)

### PR DESCRIPTION
**Pull request by iankronquist**
Fixes #595 on GitHub.

### CODE REVIEW

File: src/unix/freebsd.c [hunk 1]
```diff
@@ -176,7 +176,7 @@ char** uv_setup_args(int argc, char** argv) {
 int uv_set_process_title(const char* title) {
   int oid[4];
 
-  if (process_title) uv__free(process_title);
+  uv__free(process_title);
```
Comment (deleisha) [inline src/unix/freebsd.c]: It is usually good idea to nullify after free.

File: src/unix/openbsd.c [hunk 2]
```diff
@@ -161,7 +161,7 @@ char** uv_setup_args(int argc, char** argv) {
 
 
 int uv_set_process_title(const char* title) {
-  if (process_title) uv__free(process_title);
+  uv__free(process_title);
```
Comment (deleisha) [inline src/unix/openbsd.c]: ditto

File: src/win/getaddrinfo.c [hunk 3]
```diff
@@ -219,9 +217,7 @@ void uv_freeaddrinfo(struct addrinfo* ai) {
   char* alloc_ptr = (char*)ai;
 
   /* release copied result memory */
-  if (alloc_ptr != NULL) {
-    uv__free(alloc_ptr);
-  }
+  uv__free(alloc_ptr);
```
Comment (deleisha) [inline src/win/getaddrinfo.c]: ditto

File: src/unix/openbsd.c (hunk 2 above)
Comment (iankronquist) [inline src/unix/openbsd.c]: Actually it's not necessary here, it's immediately reassigned.

File: src/unix/freebsd.c (hunk 1 above)
Comment (iankronquist) [inline src/unix/freebsd.c]: Actually it's not necessary here, it's immediately reassigned.

File: src/win/getaddrinfo.c (hunk 3 above)
Comment (iankronquist) [inline src/win/getaddrinfo.c]: This isn't really applicable here. alloc_ptr is the local cast of a pointer passed in which is immediately lost. We would need to pass in a `struct addrinfo** ai` to actually nullify it.

**Comment by iankronquist** [comment]
@deleisha thanks for the advice, I changed two examples to nullify the pointer after freeing it. Let me know if I missed anything or you think I'm mistaken about my comments above.

File: src/unix/dl.c [hunk 4]
```diff
@@ -67,8 +65,8 @@ const char* uv_dlerror(const uv_lib_t* lib) {
 static int uv__dlerror(uv_lib_t* lib) {
   const char* errmsg;
 
-  if (lib->errmsg)
-    uv__free(lib->errmsg);
+  uv__free(lib->errmsg);
+  lib->errmsg = NULL;
```
Comment (saghul) [inline src/unix/dl.c]: This is reassigned below, no need to do it twice.

**Comment by saghul** [comment]
LGTM with a nit. Can you please make your commit log follow the guidelines in CONTRIBUTING.md? Thanks!

**Comment by iankronquist** [comment]
@saghul Okay, I just made the commit message a little longer and fixed the nit. Anything else you would like to see?

**Comment by iankronquist** [comment]
@saghul Could you take another look at this and let me know if it's ready to go?

**Comment by saghul** [comment]
LGTM. Can you please update the commit message to the guidelines in CONTRIBUTING.md? Thanks!

**Comment by iankronquist** [comment]
@saghul I tried to follow the commit message format in the contributing doc. The one thing which is missing is a subsystem. Since this touches many subsystems it's a little hard to pin one down in particular. Should I put `maint` or something similar for the subsystem?

**Comment by saghul** [comment]
No worries, I reworded the commit message and landed it in 81072b9, thanks!

**Pull request closed without merging**